### DISCUSSION TITLE
PipeWire Multimedia Processing

### DISCUSSION THREAD

**Discussion by SubhadeepJasu** [elementary/os#552] [Ideas]
Incorporate PipeWire as audio and video processing engine. Some of the plus points of PipeWire over other currently used interfaces:

- Capture and playback of audio and video with minimal latency
- Real-time Multimedia processing on audio and video
- Multiprocess architecture to let applications share multimedia content
- Seamless support for PulseAudio, JACK, ALSA and GStreamer applications
- Sandboxed applications support. Which is good since going forward elementary app ecosystem will be flatpak based

Someone has reported improvements in Bluetooth audio quality. For me personally, I also see this as a great audio interface for Ensembles and other realtime audio applications which can run and sound great natively or from Flatpak sandbox (without any user tweaks).

**Comment by SephReed** [comment]
Just want to second this.

If you want to see a bunch of shining reviews (and a few criticisms), check out this discussion:

https://news.ycombinator.com/item?id=28490078

**Reply by SubhadeepJasu** [reply]
Interesting. In my experience, although I designed my app Ensembles (pro audio app) to use pulseaudio and still output in realtime while you can play something on firefox. This requires some good CPU and it worked most of the time. With pipewire it works even better. Like I could run 2 or 3 sound clients and then run Ensembles and it would still perform optimally.

**Reply by SephReed** [reply]
It enabled me to use my usb microphone for input, and my monitor speakers for output in Bitwig.

I still have to pretend I'm using Jack, but what used to happen is that getting Bitwig to work would break my audio for the rest of the computer until I turned off Jack and restarted pulse.  With pipewire the two modes can coexist.

**Reply by rasmus91** [reply]
Pipewire allows msbc (i believe its called) codec for the Bluetooth handsfree profile. Which allows much, much better audio quality than the codec used in pulse.

Further more, pipewire lets bluetooth devicea that support multiple profiles show up with multiple entries in the audio selector, allowing the users to easily switch from headphone to hfp when they need to use the microphone.

Finally: most apps (discord, etc) also seems to be able to switch the machine to hfp when they request the microphone qhen using pipewire.

None of these things work well with pulse

**Comment by danirabbit** [comment]
I can confirm that we're running pipewire in the latest OS 8 daily 🎉

Closing as completed :)

**Reply by SubhadeepJasu** [reply]
Awesome!!! Thank you!!!
